### PR TITLE
Add files copy to core

### PR DESCRIPTION
**Pull request by AndreaLaGrotteria**
@sdelamo still a draft since I need to merge #1547 and #1545 when are approved

### CODE REVIEW

**Comment by sdelamo** [comment]
@AndreaLaGrotteria merge master into this

**Comment by AndreaLaGrotteria** [comment]
@sdelamo done

File: buildSrc/src/main/java/io/micronaut/guides/core/FilesTransferUtility.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package io.micronaut.guides.core;
+
+import io.micronaut.core.annotation.NonNull;
+import jakarta.validation.constraints.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+
+public interface FilesTransferUtility {
+    void transferFiles(@NotNull @NonNull File outputDirectory,
```
Comment (sdelamo) [inline buildSrc/src/main/java/io/micronaut/guides/core/FilesTransferUtility.java]: probably make sense to have input first and output second

File: buildSrc/src/test/java/io/micronaut/guides/core/FilesTransferUtilityTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    File file = new File(path);
+    GuideUtils.parseGuideMetadata(file,"metadata.json", jsonSchemaProvider.getSchema(), jsonMapper).ifPresent(metadatas::add);
+
+    guideProjectGenerator.generate(outputDirectory,metadatas.get(0));
+    guideProjectGenerator.generate(outputDirectory,metadatas.get(1));
+
+    filesTransferUtility.transferFiles(outputDirectory,fileBase,metadatas.get(0));
+    filesTransferUtility.transferFiles(outputDirectory,file,metadatas.get(1));
+
+    assertTrue(new File(outputPath+"/creating-your-first-micronaut-app-gradle-groovy").exists());
```
Comment (sdelamo) [inline buildSrc/src/test/java/io/micronaut/guides/core/FilesTransferUtilityTest.java]: add spaces please 🙏. The whole framework uses spaces. 

Don't do: 

`outputPath+"/creating-your-first-micronaut-app-gradle-groovy"`

do: 

`outputPath + "/creating-your-first-micronaut-app-gradle-groovy"`

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Pull request merged**